test/demo/createNotification.inttest.ts

```diff
@@ -0,0 +1,167 @@
+        });
+
+        afterEach(async () => {
+            testApp.close();
+        });
+
+        test('No toast is shown for the created notification', async () => {
+            const note = await notifsRemote.create(testWindowIdentity, validOptions);
+            await delay(1000);
+
+            const toastWindow = await getToastWindow(testApp.identity, note.id);
+            expect(toastWindow).toBe(undefined);
+
+            await notifsRemote.clear(testWindowIdentity, note.id);
+        });
+
+        describe('When passing invalid options', () => {
+            let createPromise: Promise<Notification>;
+            beforeEach(() => {
+                // Intentionally circumventing type check with cast for testing purposes
+                createPromise = notifsRemote.create(testWindowIdentity, {id: 'invalid-notification'} as NotificationOptions);
+            });
+
+            afterEach(async () => {
+                // Cleanup the created notification (just in case it does get made)
+                await createPromise.catch(() => {});
+                await notifsRemote.clear(testWindowIdentity, 'invalid-notification');
+            });
+
+            test('An error is thrown', async () => {
+                expect(createPromise).rejects.toThrow();
+            });
+
+            test('No notification is created', async () => {
+                // We want to be sure the action has completed, but don't care about the result
+                await createPromise.catch(() => {});
+
+                // Should be no card in the center for this ID
+                const noteCards = await getCardsByNotification(testWindowIdentity.uuid, 'invalid-notification');
+                expect(noteCards.length).toBe(0);
+            });
+        });
+
+        describe('With no ID specified', () => {
+            let createPromise: Promise<Notification>;
+            beforeEach(async () => {
+                createPromise = notifsRemote.create(testWindowIdentity, validOptions);
+            });
+
+            afterEach(async () => {
+                try {
+                    const note = await createPromise;
+                    await notifsRemote.clear(testWindowIdentity, note.id);
+                } catch (e) {
+                    // do nothing
+                }
+            });
+
+            test('An ID is generated by the service and the notification is created and added to the center', async () =>{
+                await expect(createPromise).resolves;
+
+                const note = await createPromise;
+
+                expect(note).toMatchObject(validOptions);
+                expect(note.id).toMatch(/[0-9]{6,9}/); // Random 9-digit numeric string
+            });
+        });
+    });
+});
+
+// TODO: Make this a util modeled on layouts' spawn utils (SERVICE-524)
+const nextUuid = (() => {
+    let count = 0;
+    return () => 'notifications-test-app-' + (count++);
```

Comment:
Wouldn't this always return `notifications-test-app-0`?.. The local variable won't persist across separate calls to the function.

Reply:
It will persist across multiple calls within the same test (including it's lifecycle hooks). So if within one test you spawn two apps they will be called `...test-app-0` and `...test-app-1` in that order. This admittedly relies on tests not leaving windows behind when they're done, but that should always be best-practice (and we can add a teardown util later if we find it becoming a problem)

Reply:
Ahh, my bad - missed the closure.

Reply:
Oh yea, I guess it's not super clear with the single-line arrow function.